src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/OffenderService.kt

```diff
@@ -627,6 +628,20 @@ class OffenderService(
     return AuthorisableActionResult.Success(convictions)
   }
 
+  fun getCaseDetail(crn: String): AuthorisableActionResult<CaseDetail> {
+    val caseDetail = when (val caseDetailResult = apDeliusContextApiClient.getCaseDetail(crn)) {
+      is ClientResult.Success -> caseDetailResult.body
+      is ClientResult.Failure.StatusCode -> when (caseDetailResult.status) {
+        HttpStatus.NOT_FOUND -> return AuthorisableActionResult.NotFound()
+        HttpStatus.FORBIDDEN -> return AuthorisableActionResult.Unauthorised()
+        else -> caseDetailResult.throwException()
+      }
+
+      is ClientResult.Failure -> caseDetailResult.throwException()
+    }
+    return AuthorisableActionResult.Success(caseDetail)
+  }
```

Comment:
could we please update this to return CaseResult types instead? It should just be a matter of replacing `AuthorisableActionResult` with `CasResult`.

Then in the controller, instead of using the local `getSuccessEntityOrThrow` function, you can use the `extractEntityFromCasResult` function in `EntityUtils.kt`